netket/_src/symmetry/representation.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+import jax.numpy as jnp
+
+from functools import reduce
+
+from netket.utils.group import Element, FiniteGroup
+from netket.operator import DiscreteJaxOperator
+
+
+class Representation:
+    """
+    A representation of a group. For a given group, it is effectively a dictionary
+    that maps the elements of that group to their corresponding operators.
+    """
+
+    def __init__(
+        self,
+        group: FiniteGroup,
+        representation_dict: dict[Element, DiscreteJaxOperator],
+    ):
+
+        operator = next(iter(representation_dict.values()))
+        hilbert_space = operator.hilbert
+
+        for element, operator in representation_dict.items():
+            assert hilbert_space == operator.hilbert
+            assert element in group.elems
+
+        assert len(group.elems) == len(representation_dict)
+
+        self.hilbert_space = hilbert_space
+        self.group = group
+        self.representation_dict = representation_dict
+
+    def __getitem__(self, key):
+        if isinstance(key, Element):
+            return self.representation_dict[key]
+        elif isinstance(key, int):
+            return self.representation_dict[self.group[key]]
+        raise TypeError("Index should be integer or group element")
+
+    def __iter__(self):
+        return iter(self.representation_dict.items())
+
+    def get_projector(self, character_index: int) -> DiscreteJaxOperator:
```

Comment:
I am against `get_` functions (it's something people used to do in C or when wrapping C libraries because classes did not exist. No reason to use it in python).

I think this can be just `projector(character_index)